modules/top-level/plugins/default.nix

```diff
@@ -0,0 +1,55 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (import ./utils.nix lib)
+    normalizedPluginType
+    normalizePluginList
+    ;
+in
+{
+  options = {
+    finalPlugins = lib.mkOption {
+      visible = false;
+      internal = true;
+      type = lib.types.listOf normalizedPluginType;
+      description = ''
+        Final list of (normalized) plugins that will be passed to the wrapper.
+
+        It notably implements:
+        - byte-compilation (performance.byteCompileLua): in this file
+        - plugins combining (performance.combinePlugins): in ./combine-plugins.nix
+      '';
+    };
+  };
+
+  config = {
+    finalPlugins =
+      let
+        combinePlugins = import ./combine-plugins.nix { inherit lib pkgs; };
+        byteCompilePlugins = import ./byte-compile-plugins.nix { inherit lib pkgs; };
+
+        # Normalized and optionally byte compiled plugin list
+        normalizedPlugins =
+          let
+            normalized = normalizePluginList config.extraPlugins;
+          in
+          if config.performance.byteCompileLua.enable && config.performance.byteCompileLua.plugins then
+            byteCompilePlugins normalized
+          else
+            normalized;
```

Comment:
I think this would be neater with `lib.pipe`:

```nix
normalizedPlugins = lib.pipe config.extraPlugins [
  normalizePluginList
  (
    if config.performance.byteCompileLua.enable && config.performance.byteCompileLua.plugins then
      byteCompilePlugins
    else
      lib.id
  )
];
```

or

```nix
normalizedPlugins = lib.pipe config.extraPlugins (
  [
    normalizePluginList
  ]
  ++ lib.optionals (config.performance.byteCompileLua.enable && config.performance.byteCompileLua.plugins) [
    byteCompilePlugins
  ]
);
```

---

Also: maybe we should have a `cfg`-like binding for `config.performance.byteCompileLua`? Maybe `byteCompCfg = config.performance.byteCompileLua`?